lib/cluster.rb

```diff
@@ -169,11 +177,31 @@ def add_members_from(cluster)
     end
   end
 
+  def commitments?(include_deprecated: false)
+    if include_deprecated
+      commitments.any?
+    else
+      commitments.reject(&:deprecated?).any?
+    end
+  end
+
+  def any_current_holdings?
+    @any_current_holdings ||= ch_counts.values.sum > 0
+  end
+
+  def eligible_for_commitments?
+    # be an spm with items and 1+ CH holdings
+    @eligible_for_commitments ||= format == "spm" && ht_items.any? && any_current_holdings?
```

Comment:
I had written a comment about this, but I think it's fine as-is -- I can imagine a case where we might want to get a report of eligible clusters without finding the specific holdings; if we do find the specific holdings it's just one more iteration over the holdings (not N more, since this is memoized); and this should short-circuit if it's not an spm or there aren't any HTItems.